library/kotlin/test/io/envoyproxy/envoymobile/RequestMapperTest.kt

```diff
@@ -0,0 +1,211 @@
+
+  @Test
+  fun `request headers are forwarded to outbound request headers`() {
+    val requestHeaders = RequestBuilder(method = RequestMethod.POST, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .addHeader("header_1", "value_a")
+      .addHeader("header_2", "value_b")
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders["header_1"]).isEqualTo("value_a")
+    assertThat(requestHeaders["header_2"]).isEqualTo("value_b")
+  }
+
+  @Test
+  fun `retry policy is added to outbound request headers`() {
+    val retryPolicy = RetryPolicy(
+      maxRetryCount = 123,
+      retryOn = listOf(
+        RetryRule.STATUS_5XX,
+        RetryRule.GATEWAY_ERROR,
+        RetryRule.CONNECT_FAILURE,
+        RetryRule.RETRIABLE_4XX,
+        RetryRule.REFUSED_UPSTREAM),
+      perRetryTimeoutMs = 9001)
+    val retryPolicyHeaders = retryPolicy.outboundHeaders()
+
+    val requestHeaders = RequestBuilder(method = RequestMethod.POST, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .addRetryPolicy(retryPolicy)
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders).containsAllEntriesOf(retryPolicyHeaders)
+  }
+
+  @Test
+  fun `retry policy headers are not overwritten in outbound request headers`() {
+    val retryPolicy = RetryPolicy(
+      maxRetryCount = 123,
+      retryOn = listOf(
+        RetryRule.STATUS_5XX,
+        RetryRule.GATEWAY_ERROR,
+        RetryRule.CONNECT_FAILURE,
+        RetryRule.RETRIABLE_4XX,
+        RetryRule.REFUSED_UPSTREAM),
+      perRetryTimeoutMs = 9001)
+    val retryPolicyHeaders = retryPolicy.outboundHeaders()
+
+    val requestHeaders = RequestBuilder(method = RequestMethod.POST, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .addHeader("x-envoy-max-retries", "override")
+      .addRetryPolicy(retryPolicy)
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders).containsAllEntriesOf(retryPolicyHeaders)
+  }
+
+  @Test
+  fun `delete method is added to outbound request headers`() {
+    val requestHeaders = RequestBuilder(method = RequestMethod.DELETE, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders[":method"]).isEqualTo("DELETE")
+  }
+
+  @Test
+  fun `get method is added to outbound request headers`() {
+    val requestHeaders = RequestBuilder(method = RequestMethod.GET, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders[":method"]).isEqualTo("GET")
+  }
+
+  @Test
+  fun `head method is added to outbound request headers`() {
+    val requestHeaders = RequestBuilder(method = RequestMethod.HEAD, scheme = "https", authority = "api.foo.com", path = "/foo")
+      .build()
+      .outboundHeaders()
+
+    assertThat(requestHeaders[":method"]).isEqualTo("HEAD")
+  }
+
+  @Test
+  fun `options method is added to outbound request headers`() {
```

Comment:
Not sure these additional tests for each of the `RequestMethod` cases really provide much value, but no harm in keeping them

Reply:
I accidentally messed up writing the mapping from `POST` to `"POST"`

I mean the code review would've caught that but a test allowed me to not embarrass myself